static/app/components/events/searchBar.tsx

```diff
@@ -33,63 +33,75 @@ const SEARCH_SPECIAL_CHARS_REGEXP = new RegExp(
   'g'
 );
 
-const getFunctionTags = (fields: Readonly<Field[]>) =>
-  Object.fromEntries(
-    fields
-      .filter(
-        item =>
-          !Object.keys(FIELD_TAGS).includes(item.field) &&
-          !isEquation(item.field) &&
-          !isCustomMeasurement(item.field)
-      )
-      .map(item => [
-        item.field,
-        {key: item.field, name: item.field, kind: FieldKind.FUNCTION},
-      ])
-  );
+const STATIC_FIELD_TAGS_SET = new Set(Object.keys(FIELD_TAGS));
+const getFunctionTags = (fields: Readonly<Field[]> | undefined) => {
```

Comment:
@k-fish I think this should yield the most improvement (assumings fields has some large N here) as before it was doing Object.keys + includes for each item it was iterating over

Reply:
oh so maybe O(n^2) or O(n*m)? That could do it 🤔. 10000 items should take trivial amounts of time to iterate over as long as they aren't huge and are monomorphic.

Reply:
Yeah, n*m here, but for the Object.entries + map I suspect that is On^2

Reply:
I just put up a quick benchmark - https://jsbench.me/msl9ynj2sc/1 - I get about 70% slower run on fromEntries